sway/layout.c

```diff
@@ -192,6 +201,33 @@ void arrange_windows(swayc_t *container, int width, int height) {
 		}
 		break;
 	}
+
+	// Arrage floating layouts for workspaces last
+	if (container->type == C_WORKSPACE) {
+		for (i = 0; i < container->floating->length; ++i) {
+			swayc_t *view = ((swayc_t *)container->floating->items[i]);
+			// Set the geometry
+			struct wlc_geometry geometry = {
+				.origin = {
+					.x = view->x,
+					.y = view->y
+				},
+				.size = {
+					.w = view->width,
+					.h = view->height
+				}
+			};
+			wlc_view_set_geometry(view->handle, &geometry);
+
+			// Bring the views to the front in order of the list, the list
+			// will be kept up to date so that more recently focused views
+			// have higher indexes
+			// This is conditional on there not being a fullscreen view in the workspace
+			if (!(wlc_view_get_state(container->focused->handle) & WLC_BIT_FULLSCREEN)) {
```

Comment:
Not necessary for merging this PR, imo, but feel free to investigate it if you like.